checkov/terraform/graph_builder/foreach_handler.py

```diff
@@ -216,21 +224,21 @@ def _create_new_resource(self, main_resource: TerraformBlock, new_value: int | s
 
         idx_to_change = new_key or new_value
         self._add_index_to_block_properties(new_resource, idx_to_change)
-        self.local_graph.vertices.append(new_resource)
+        if resource_idx is not None:
+            self.local_graph.vertices[resource_idx] = new_resource
```

Comment:
This logic is a bit scary in my opinion, as it basically relies on the fact that you need to pass the correct input to this function in order for it to work, meaning sending `resoucre_idx = None` or not pass it at all.
Instead you can move line 161 to be just here, so the logic of assigning `None` to `resource_idx` will be part of the function itself and not separated from it.